presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlIntegrationSmokeTest.java

```diff
@@ -151,6 +151,22 @@ public void testMySqlTinyint1()
         assertUpdate("DROP TABLE mysql_test_tinyint1");
     }
 
+    @Test
+    public void testCharTrailingSpace()
```

Comment:
you could also add test with `deprecated.legacy-char-to-varchar-coercion` set to `true`

Reply:
This test is being run with deprecated.legacy-char-to-varchar-coercion set to true because true is the default (which it seems like we both agreed on in your previous comment).  Did you mean false?

After about an hour of looking at various tests, I'm not exactly sure how to set a config setting for a particular test.  If you want me to do this, can you tell me how, or point me to a relevant example?

Reply:
> This test is being run with deprecated.legacy-char-to-varchar-coercion set to true because true is the default (which it seems like we both agreed on in your previous comment).

By default `deprecated.legacy-char-to-varchar-coercion` is set to `false` on master.

> After about an hour of looking at various tests, I'm not exactly sure how to set a config setting for a particular test. If you want me to do this, can you tell me how, or point me to a relevant example?

I agree it is difficult. We can skip tests, but please make JDBC connector fail when there is char predicate and `deprecated.legacy-char-to-varchar-coercion` is set to `true`.

Reply:
> I agree it is difficult and I have double checked that there is no nice way to provide such parameters only for singular test case.

you could create an adhoc query runner within test method. You can add `Map<String, String> extraProperties` parameter
to `com.facebook.presto.plugin.postgresql.PostgreSqlQueryRunner#createPostgreSqlQueryRunner` and pass it down to `DistributedQueryRunner`, thus letting you control FeatureConfig.

Reply:
> you could create an adhoc query runner within test method

but this will instantiate database instance, no?

Reply:
> but this will instantiate database instance, no?

yes, it will

> It looks like the createPostgreSqlQueryRunner is actually being called in a super, and is the same query runner for all the tests. So I can't have some tests run with the option, and some without.

you can, but only if you create a separate runner for your special tests. 
we do this sometimes in other tests -- eg https://github.com/prestodb/presto/blob/bd96c0f2ad4efae04ae6b61c1fcb1581e74a7a07/presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java#L890 
(of course, `FunctionAssertions` uses a lightweight `LocalQueryRunner`; creating new PostgreSqlQueryRunner needs to be limited only to the case where necessary)

> If I make a function that creates another DistributedQueryRunner, then it says that the postgres schema doesn't exist. I'm guessing this makes a new database instance or doesn't connect to the same one?

can you share the code so that there is something to be looked at? 
eg you can push this non-working version as an update to this PR


anyway i am actually fine with not having a separate test with different value of the toggle -- as i expressed in https://github.com/prestodb/presto/pull/12104#issuecomment-450993985. This is perhaps only because I don't appreciate the potential nuances @sopel39 is concerned about.

Reply:
Actually, give me a couple minutes, I think I have figured out what was going wrong (it had to do with not setting up the connection string correctly.  I think I've got something working that is local, and can just run a few tests with this setting set differently.

Reply:
Pushed, hopefully these tests satisfy.